pkg/collector/adapters/config_to_ports.go

```diff
@@ -60,32 +61,46 @@ func ConfigToReceiverPorts(logger logr.Logger, config map[interface{}]interface{
 		return nil, ErrReceiversNotAMap
 	}
 
+	sortedNames := make([]string, 0, len(receivers))
+	for receiverName := range receivers {
+		name, ok := receiverName.(string)
+		if !ok {
+			logger.V(2).Info("receiver does not match type, want: %s, got: %T", "string", receiverName)
+			continue
+		}
+		sortedNames = append(sortedNames, name)
+	}
+	sort.Strings(sortedNames)
+
 	ports := []corev1.ServicePort{}
-	for key, val := range receivers {
+	for _, name := range sortedNames {
 		// This check will pass only the enabled receivers,
 		// then only the related ports will be opened.
-		if !recEnabled[key] {
+		if !recEnabled[name] {
 			continue
 		}
-		receiver, ok := val.(map[interface{}]interface{})
+
+		receiver, ok := receivers[name].(map[interface{}]interface{})
 		if !ok {
-			logger.Info("receiver doesn't seem to be a map of properties", "receiver", key)
+			logger.Info("receiver doesn't seem to be a map of properties", "receiver", name)
 			receiver = map[interface{}]interface{}{}
 		}
 
-		rcvrName := key.(string)
-		rcvrParser := parser.For(logger, rcvrName, receiver)
+		rcvrParser := parser.For(logger, name, receiver)
 
 		rcvrPorts, err := rcvrParser.Ports()
 		if err != nil {
 			// should we break the process and return an error, or just ignore this faulty parser
 			// and let the other parsers add their ports to the service? right now, the best
 			// option seems to be to log the failures and move on, instead of failing them all
-			logger.Error(err, "parser for '%s' has returned an error: %w", rcvrName, err)
+			logger.Error(err, "parser for '%s' has returned an error: %w", name, err)
 			continue
 		}
 
 		if len(rcvrPorts) > 0 {
+			sort.Slice(ports, func(i, j int) bool {
```

Comment:
From all changes in this file only these 3 lines are needed 101-103 and move then right before the return.